File: Android/scenes/ktv_singbattle/src/main/java/io/agora/scene/ktv/singbattle/widget/song/OnSongActionListener.java
package io.agora.scene.ktv.singbattle.widget.song;

import androidx.annotation.NonNull;

public interface OnSongActionListener {

    /**
     * Choose song - callback when pull down to refresh
     */
    void onChooseSongRefreshing(@NonNull SongDialog dialog);

    /**
     * Choose song - callback when item "choose song" button clicked
     */
    void onChooseSongChosen(@NonNull SongDialog dialog, @NonNull SongItem songItem);

    /**
     * Chosen song list - callback when item "delete" clicked
     */
    void onChosenSongDeleteClicked(@NonNull SongDialog dialog, @NonNull SongItem song);

    /**
     * Chosen song list - callback when item "top" clicked
     */
    void onChosenSongTopClicked(@NonNull SongDialog dialog, @NonNull SongItem song);

    void onStartSingBattleGame(@NonNull SongDialog dialog);
}

File: Android/scenes/cantata/src/main/java/io/agora/scene/cantata/widget/song/OnSongActionListener.kt
package io.agora.scene.cantata.widget.song

interface OnSongActionListener {
    /**
     * Song selection - Callback when pull-to-refresh
     */
    fun onChooseSongRefreshing(dialog: SongDialog)

    /**
     * Song selection - Callback when "Choose Song" button is clicked on item
     */
    fun onChooseSongChosen(dialog: SongDialog, songItem: SongItem)

    /**
     * Selected song list - Callback when "Delete" is clicked on item
     */
    fun onChosenSongDeleteClicked(dialog: SongDialog, song: SongItem)

    /**
     * Selected song list - Callback when "Top" is clicked on item
     */
    fun onChosenSongTopClicked(dialog: SongDialog, song: SongItem)
}